bookish/Services/auxiliary_functions.py

```diff
@@ -0,0 +1,31 @@
+import hashlib
+from bookish.models.users import Users
+
+
+def hashing_algorithm(string):
+    result = hashlib.md5(string.encode('utf-8')).hexdigest()
+    return result
+
+
+def check_auth_token(token):
+    """
+    :param token:
+    :return: true -> token valid
+             false -> token expired/invalid
+    """
+
+    all_users = Users.query.all()
+    for user in all_users:
+        print(user)
+        if user.user_token == token and user.user_token_expire > datetime.datetime.now():
+            return True
+    return False

```

Comment:
Can't we rethink this to send the user as a parameter instead of iterating through all the users? It seems like unnecessary computation? 

Maybe just add the `user name` as part of the token.

Reply:
Could we add the name to the token and then use next() to find the user?

Reply:
Yes, sounds good 😄